cdap-ui/app/cdap/components/Footer/index.js

```diff
@@ -17,16 +17,45 @@
 import React from 'react';
 import { Theme } from 'services/ThemeHelper';
 import If from '../If';
-import { objectQuery } from 'services/helpers';
+import { isNilOrEmptyString, objectQuery } from 'services/helpers';
+import NamespaceStore from 'services/NamespaceStore';
 require('./Footer.scss');
 
 export default function Footer() {
   const footerText = Theme.footerText;
   const footerUrl = Theme.footerLink;
   // 'project-id-30-characters-name1/instance-id-30-characters-name';
   const instanceMetadataId = objectQuery(window, 'CDAP_CONFIG', 'instanceMetadataId');
+  const [selectedNamespace, setSelectedNamespace] = React.useState(
+    NamespaceStore.getState().selectedNamespace
+  );
+  React.useEffect(() => {
+    const sub = NamespaceStore.subscribe(() =>
+      setSelectedNamespace(NamespaceStore.getState().selectedNamespace)
+    );
+    const mutationObserver = new MutationObserver((mutations) => {
+      if (!Array.isArray(mutations) || mutations.length === 0) {
+        return;
+      }
+      const title = objectQuery(mutations[0], 'target', 'textContent');
+      if (isNilOrEmptyString(title)) {
+        return;
+      }
+      const featureName = (objectQuery(title.split('|'), 1) || '').trim();
+      // For non-namespace pages: dashboard and admin.
+      if ((featureName == 'Operations') | (featureName == 'Namespace')) {
```

Comment:
```
// Outside of the component can define a constant
const nonNamespacePages = ['Operations', 'Namespace'];
export default function Footer() {
  ...
  if (nonNamespacePages.indexOf(featureName) !== -1) {
    setSelectedNamespace('--');
  }
}

```

Reply:
There's actually quite a bit of pages that are not namespaced:
- Operations
- Reports
- Admin
- Http caller
- Profiles (Create and Detail)

the featureName Namespace is actually namespaced.